src/status_im/contexts/settings/wallet/saved_addresses/add_address_to_save/view.cljs

```diff
@@ -161,7 +161,9 @@
                                                (rf/dispatch
                                                 [:open-modal :screen/settings.save-address]))
                                              [address ens-name? address-or-ens])]
-    (rn/use-unmount #(rf/dispatch [:wallet/clear-address-to-save]))
+    (rn/use-mount (fn []
+                    (rf/dispatch [:wallet/clean-scanned-address])
```

Comment:
One small addition: I would suggest dispatching the same event before we take the user to the scanner page (in `on-scan-address`)

Reply:
@smohamedjavid Done